Documentation/design-docs/COM-activation.md

```diff
@@ -0,0 +1,92 @@
+
+```
+"Assembly"="mscorlib, Version=1.0.5000.0, Culture=neutral, PublicKeyToken=b77a5c561934e089"
+"Class"="System.TypeLoadException"
+"RuntimeVersion"="v1.1.4322"
+```
+
+The above registration is typically done with the [`RegAsm.exe`](https://docs.microsoft.com/en-us/dotnet/framework/tools/regasm-exe-assembly-registration-tool) tool. Alternatively, registry scripts can be generated by `RegAsm.exe`.
+
+### .NET Core Class COM Activation
+
+In .NET Core, our intent will be to avoid a system wide shim library. This may add additional cost for deployment scenarios, but will reduce servicing and engineering costs by making deployment more explicit and less magic.
+
+The current hosting solutions are defined at [Documentation/design-docs/host-components.md](https://github.com/dotnet/core-setup/blob/master/Documentation/design-docs/host-components.md). Along with the existing hosts an additional COM activation host library will be added. This library will export the required functions for COM class activation.
+
+>[`HRESULT DllGetClassObject(REFCLSID rclsid, REFIID riid, LPVOID *ppv);`](https://docs.microsoft.com/en-us/windows/desktop/api/combaseapi/nf-combaseapi-dllgetclassobject)
+
+>[`HRESULT DllCanUnloadNow();`](https://docs.microsoft.com/en-us/windows/desktop/api/combaseapi/nf-combaseapi-dllcanunloadnow)
+
+When `DllGetClassObject()` is called in an activation scenario, the following will occur:
+
+1) Determine additional registration information needed for activation.
+    * **The location and mechanics of accessing this registration information is still TBD**
+    * `Assembly` (**required**) - the [Fully-Qualified Name](https://docs.microsoft.com/en-us/dotnet/framework/reflection-and-codedom/specifying-fully-qualified-type-names) of the assembly.
+    * `Class` (**required**) - the full type name (e.g. `MyCompany.MyProduct.MyClass`).
+    * `RuntimeVersion` (optional) - if not supplied the latest CLR found will be used.
+    * `Codebase` (optional) - an absolute path to the assembly to load. If not supplied, the default [`AssemblyLoadContext`](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.loader.assemblyloadcontext?view=netcore-2.1) will be used to attempt to find the assembly. If provided, a [`runtimeconfig.json`](https://github.com/dotnet/cli/blob/master/Documentation/specs/runtime-configuration-file.md) file adjacent to the assembly can be used to describe runtime configuration details.
+1) Using the existing `hostfxr` library, attempt to discover the desired CLR.
+    * A current running CLR will be preferred, otherwise a new CLR instance will be created.
+1) A request to the CLR via [`Activator.CreateInstance()`](https://docs.microsoft.com/en-us/dotnet/api/system.activator.createinstance?view=netcore-2.1) will be used to instantiate an instance of the type in the assembly.
```

Comment:
What if there are multiple COM components in the same process and each come with different dependencies (e.g. each comes with different version of Json.NET)? We may need to load each COM component into its own AssemblyLoadContext to handle this.

Reply:
>  We may need to load each COM component into its own AssemblyLoadContext to handle this.

That was my intent - I will call this out explicitly.